src/ZFTool/Controller/DiagnosticsController.php

```diff
@@ -224,13 +256,76 @@ public function runAction()
             } else {
                 $model->setErrorLevel(0);
             }
-        } else {
+            return $model;
+    }
+
+    private function processHttpRequest(Request $request, Collection $results)
+    {
+        $defaultAccept = new Accept();
+        $defaultAccept->addMediaType(self::CONTENT_TYPE_HTML);
+
+        $acceptHeader = $request->getHeader('Accept', $defaultAccept);
+
+        $viewModel = function () use ($results) {
+            return new ViewModel(array(
+                'results' => $results,
+            ));
+        };
+
+        if ($acceptHeader->match(self::CONTENT_TYPE_HTML)) {
```

Comment:
If it isn't JSON, then it's either HTML or something else. In both those cases, you return the `ViewModel` instance.